### PR TITLE
Knife refactor and minor changes regarding.

### PR DESCRIPTION
**Pull request by Mechoid**
Knives become actual knives, hatchets swapped over to be a subtype of machete, which is a subtype of knife that cleaves. Towercap logs can be cut by any sharp or edged weapon.

### CODE REVIEW

**Comment by Anewbe** [comment]
Does this remove functionality from any of the knives that used to be hatchet-based?

**Comment by Mechoid** [comment]
Theoretically, only the chopping of tower-caps would be changed, due to them looking for an actual hatchet instead, though they gain all the functionality of normal knives, including ghetto surgery. Actual logs check for sharp or edge I believe, which they still have.

File: code/modules/surgery/generic.dm [hunk 1]
```diff
@@ -303,7 +303,7 @@
 /datum/surgery_step/generic/amputate
 	allowed_tools = list(
 	/obj/item/weapon/surgical/circular_saw = 100, \
-	/obj/item/weapon/material/hatchet = 75
+	/obj/item/weapon/material/knife/machete/hatchet = 75
```
Comment (elgeonmb) [inline code/modules/surgery/generic.dm]: allow non-hatchet machetes to be used as csaws?

Comment (Anewbe) [reply code/modules/surgery/generic.dm]: Nah, hatchet makes sense, IMO, regular machete does not.

**Pull request merged**